cfme/networks/security_group.py

```diff
@@ -54,7 +54,16 @@ def all(self):
         else:
             view = navigate_to(self, 'All')
         list_networks_obj = view.entities.get_all(surf_pages=True)
-        return [self.instantiate(name=s.name) for s in list_networks_obj]
+
+        collection = (self.instantiate(name=s.name) for s in list_networks_obj)
+        return list(a for a in collection if apply_filters(a, self.filters))
+
+
+def apply_filters(item, filters):
+    for attr_name, val in filters.items():
+        if getattr(item, attr_name) != val:
```

Comment:
Question:
I found `SecurityGroup` main attribute is `name`. You are also trying to filter it with a name. 
Why are you trying to implement this separate method?
Is security group allow to create duplicate names?

Reply:
1) I do not know what is meant by main attribute.

2) `apply_filters` is written so it is general -- can be used with any object, therefore it doesn't belong to any specific class. Maybe it could be in some base class though. Best would be to have it in some file and import it when needed.